maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/PluginReport.java

```diff
@@ -197,9 +197,18 @@
     @Component
     private RuntimeInformation rtInfo;
 
+    /**
+     * Path to {@code plugin.xml} plugin descriptor to generate the report from.
+     *
+     * @since 3.5.1
+     */
+    @Parameter( defaultValue = "${project.build.outputDirectory}/META-INF/maven/plugin.xml", required = true )
```

Comment:
I would like to see this as a readOnly parameter. Changing its value will break things.

Reply:
Which things will it break?

Reply:
https://maven.apache.org/ref/3.5.2/maven-plugin-api/ says that the `plugin.xml` is alwys expected at this location. So it should be considered a contant, not a variable, hence why I would like to see it as readOnly.

Reply:
Thanks for the explanation, @rfscholte . But https://maven.apache.org/ref/3.5.2/maven-plugin-api/ seems to define the the location of the `plugin.xml` within the `jar` rather than its location in the source tree. This `pluginXmlFile` indeed is a location in the source tree and this `PluginReport` is a report mojo that will not impact the location of `plugin.xml` in the jar in any way. That said, I cannot see how a writable `pluginXmlFile` can violate the maven-plugin-api.

Reply:
The claim of breaking things is a bit too hard, but making it variable assumes that is configurable in real world. It will also appear on reports like the goal page, the help goal, etc.
Let's not confuse people or show parameters which will only distract them.
What is the issue of *not* making it readOnly?